GVFS/GVFS.Common/Git/GitRepo.cs

```diff
@@ -44,14 +44,22 @@ private enum LooseBlobState
             Unknown,
         }
 
-        public bool HasActiveLibGit2Repo => this.libgit2RepoInvoker?.IsActive == true;
-
         public GVFSLock GVFSLock
         {
             get;
             private set;
         }
 
+        public void CloseActiveRepo()
+        {
+            this.libgit2RepoInvoker?.DisposeSharedRepo();

```

Comment:
IIUC - this can be called after the `GitRepo` has been disposed, and that is why we are using the `?.` operator here? Assuming my interpretation is correct, it is a bit subtle as to why we we use this pattern here, but not elsewhere in the file - maybe a small comment to let future readers know why this is.

Reply:
The unit tests for PackfileMaintenanceStep use the constructor that leaves this `null`. That's the only reason.